WordPress/WordPressTest/CommentService+LikesTests.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Nimble
+import OHHTTPStubs
+import XCTest
+
+@testable import WordPress
+
+final class CommentService_LikesTests: CoreDataTestCase {
+
+    private var commentService: CommentService!
+
+    override func setUp() {
+        super.setUp()
+        commentService = CommentService(coreDataStack: contextManager)
+    }
+
+    override func tearDown() {
+        HTTPStubs.removeAllStubs()
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_likeComment_givenSuccessfulAPICall_updateLikesCount() {
+        let post = ReaderPost(context: mainContext)
+        post.siteID = 1
+        post.postID = 2
+        let comment = Comment(context: mainContext)
+        comment.commentID = 3
+        comment.post = post
+        contextManager.saveContextAndWait(mainContext)
+
+        // Add a successful HTTP API call stub
+        stub(condition: isPath("/rest/v1.1/sites/1/comments/3/likes/new")) { _ in
+            HTTPStubsResponse(
+                jsonObject: [String: Any](),
+                statusCode: 200,
+                headers: nil
+            )
+        }
+
+        // Call the toggle like function and wait for it to complete
+        waitUntil { done in
+            self.commentService.toggleLikeStatus(for: comment, siteID: 1) {
+                done()
+            } failure: { error in
+                XCTFail("Unexpected error: \(String(describing: error))")
+                done()
+            }
+        }
+
+        // The comment's like status should be changed
+        expect(comment.isLiked).toEventually(beTrue())
+        expect(comment.likeCount).toEventually(equal(1))
```

Comment:
Do these checks need to be async (`.toEventually`) or can they be sync (`.to`)? I mean, do we need to wait further for `comment` to change after having already waited for `toggleLikeStatus...` to completed?

I the answer is no, what do you think of moving the check in the completion closure?

```swift
waitUntil { done in
    self.commentService.toggleLikeStatus(for: comment, siteID: 1) {
        expect(comment.isLiked).toEventually(beTrue())
        expect(comment.likeCount).toEventually(equal(1))
    } failure: { error in
        XCTFail...
        // don't even need to call done, since we already made the test fail
```

Reply:
> In this particular instance, I think `XCTAssert` to make assertion immediately would work too

_Just curious_. Why did you decide to use `XCTAssertTrue(foo)` instead of `expect(foo).to(beTrue())` and `XCTAssertEqual(foo, 1))` instead of `expect(foo) == 1`?

Personally, I like the look of `expect` more, but that's not a good reason to recommend it 🙃 I guess the benefit of `XCTAssert...` is that it's 1st party.

Reply:
To be honest, I didn't even think about choosing between the two 😄 . I could easily go with the `expect` one in an alternative universe.

However, I do like the helper functions in Nimble like the `toEvetually` and the `waitUtil` functions. I think you can do the same using `XCTestExpectation`, but it would be much more verbose.

Reply:
No worries. I didn't mean that to come across as a request to translate to Nimble. However, I appreciate it. It might just be me, but it makes more sense to see Nimble expectations used together with Nimble `waitUntil`.

🙇‍♂️